app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume_snapshot.rb

```diff
@@ -81,7 +81,17 @@ def delete_snapshot_queue(userid = "system", _options = {})
   def delete_snapshot(_options = {})
     with_provider_object do |snapshot|
       if snapshot
-        snapshot.delete
+        begin
+          snapshot.delete
+        rescue => e
+          notification_options = {
+            :snapshot_op => 'delete',
+            :subject     => "[#{name}]",
+            :error       => e.to_s
+          }
+
+          Notification.create(:type => :vm_snapshot_failure, :options => notification_options)
```

Comment:
I think the problem here is that you are actually hiding the exception from the caller. Previously, in case of an error, this was propagated up the call stack. Now, only the notification is thrown, and while this will resolve the issue we had in the UI, it does not handle the case when this function is called somewhere else.

Raise this error: https://github.com/ManageIQ/manageiq-gems-pending/blob/476478cb5aaec1139f052269a9756e0fcd7c4668/lib/gems/pending/util/miq-exception.rb#L130.

Here's an example I've shown and also raises the error.